### ISSUE TITLE
Server vs. Client Routing: Which Takes Precedence When URL Paths Match?

### ISSUE THREAD

**Issue by bkyerv** [szymmis/vite-express#89]
Am I correct in understanding that if the URL path matches any server-side endpoint, that endpoint will be served, even if there's also a matching client-side route for the same path?

**Comment by szymmis** [comment]
Hi @bkyerv, you are absolutely right. The middleware that serves HTML is injected at the end of the chain, so the order in which path is resolved is: 

- first handlers registered on the server are considered
- then if none of them closed the request, vite-express serves HTML and static files. 

You can manipulate when static files are served by using `static` middleware, if you have some kind of auth middleware that would block static file requests for example.

Basically, serving HTML acts as a fallback to when server didn't fully handled the request, so you could say that there is always a maching endpoint for every endpoint that exist or might exist.

Let me know if that fully answers your question, and ask if something is still unclear!

**Comment by szymmis** [comment]
Two weeks have passed since my response so I assume that it answered your questions.
I'll proceed and close this issue. As always, if you have any more questions feel free to continue this thread.
